### PR TITLE
41 add tests

### PR DESCRIPTION
**Pull request by blakethomp**
Merge tests for socrata and socrata_filter modules.

### CODE REVIEW

File: src/Entity/Endpoint.php [hunk 1]
```diff
@@ -1,8 +1,4 @@
 <?php
-/**
```
Comment (PapaGrande) [inline src/Entity/Endpoint.php:2]: Why was this removed?

File: config/schema/socrata.schema.yml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 socrata.settings:
-  type: mapping
```
Comment (PapaGrande) [inline config/schema/socrata.schema.yml:2]: Can you explain this change?

File: socrata_filter/tests/src/Kernel/FilterSocrataTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $data = [
+      'url' => 'https://data.seattle.gov/resource/rn6u-vkuv.json',
+      'id' => 'endpoint',
+      'label' => 'Endpoint',
+      'app_token' => '',
+    ];
```
Comment (PapaGrande) [inline socrata_filter/tests/src/Kernel/FilterSocrataTest.php]: Would it make sense to put this info into an easily-editable generic YML file for use by all tests?

Like https://www.drupal.org/docs/8/creating-custom-modules/add-a-default-configuration.

Then we can have instances such as "goodURL" and "badURL" for testing.

File: tests/src/Unit/EndpointTest.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+  /**
+   * Test getting the URL components.
+   *
+   * The dataset is the string of characters followed by the .json file
+   * extension.
+   */
+  public function testGetComponents() {
+    $components = $this->endpoint->getComponents();
+    $this->assertEquals('rn6u-vkuv', $components['dataset_id']);
```
Comment (PapaGrande) [inline tests/src/Unit/EndpointTest.php]: This could be parsed out of the URL with RegEx instead of being hardcoded.

**Review by PapaGrande — CHANGES_REQUESTED** [review]

File: src/Entity/Endpoint.php (hunk 1 above)
Comment (blakethomp) [reply src/Entity/Endpoint.php:2]: Code sniffer said namespaced classes should not start with a file doc comment.

File: config/schema/socrata.schema.yml (hunk 2 above)
Comment (blakethomp) [reply config/schema/socrata.schema.yml:2]: It was my understanding that this YAML was for a configuration object, which has a subtype that should be used

> There are two types that configuration schema predefines for configuration files: config_object for global configuration files and config_entity for entities.

https://www.drupal.org/docs/8/api/configuration-api/configuration-schemametadata

File: socrata_filter/tests/src/Kernel/FilterSocrataTest.php (hunk 3 above)
Comment (blakethomp) [reply socrata_filter/tests/src/Kernel/FilterSocrataTest.php]: I moved config to a YAML file in a /tests/config dir in the main module. I don't think the /config/install dir would be appropriate, nor using the config API, I think that's really for module settings configuration.

**Review by PapaGrande — APPROVED** [review]

**Comment by PapaGrande** [comment]
@blakestein Can you resolve the merge conflicts?

**Pull request merged**